### DISCUSSION TITLE
Need info about how WSL2 boots

### DISCUSSION THREAD

**Discussion by GutiPerez** [microsoft/WSL#8014] [Q&A]
Hello!

I would like to know where I could find information about the process of booting a distro on WSL2. My problem is that I have tried importing a custom distro to WSL2, but all I get is the file system mount with a root user. I am trying to figure out how I should configure my custom distro so that the first time it is launched it folllows a specific process (like doing some configurations, creating an user, etc.). I know, for example, that _systemmd_ is not available on WSL2, so I guess launching a distro from WSL2 has some peculiarities that need to be addressed.

So, to sum up, where could I find more information about the boot process on WSL2 or some guidance about how I should configure my distro so that it boots like on a normal environment but on WSL2.

Thanks in advance.
